pkg/monitor/monitor/monitor_manager.go

```diff
@@ -370,10 +370,23 @@ func (m *MonitorManager) syncTidbMonitorConfig(monitor *v1alpha1.TidbMonitor) (*
 			return nil, err
 		}
 		if externalContent, ok := externalCM.Data["prometheus-config"]; ok {
-			newCM.Data["prometheus-config"] = externalContent
+			newCM.Data["prometheus.yml"] = externalContent
 		}
 	}
-	return m.deps.TypedControl.CreateOrUpdateConfigMap(monitor, newCM)
+	_, err = m.deps.TypedControl.CreateOrUpdateConfigMap(monitor, newCM)
+	if err != nil {
+		klog.Errorf("tm[%s/%s]'s configMap failed to createOrUpdate,err: %v", newCM.Namespace, newCM.Name, err)
+		return nil, err
+	}
+	if monitor.Spec.Grafana != nil {
+		grafanaCM := getGrafanaConfigMap(monitor)
+		_, err = m.deps.TypedControl.CreateOrUpdateConfigMap(monitor, grafanaCM)
+		if err != nil {
+			klog.Errorf("tm[%s/%s]'s configMap failed to createOrUpdate,err: %v", newCM.Namespace, grafanaCM.Name, err)
+			return nil, err
+		}
+	}
+	return nil, err
```

Comment:
The CM returned here will also be used by `getMonitorStatefulSet`, how can you just return nil here?